### PR TITLE
get_army_result now takes the most reported result

### PR DESCRIPTION
**Pull request by Rackover**
New version of https://github.com/FAForever/server/pull/262 that replaces #257 
Credit to @micheljung  or @Crotalus , I'm not sure. 
Closing theirs.

### CODE REVIEW

**Comment by Rackover** [comment]
Also, fixes https://github.com/FAForever/server/issues/367

File: server/games/game.py [hunk 1]
```diff
@@ -711,7 +711,14 @@ def get_army_score(self, army):
         return score
 
     def get_army_result(self, player):
-        return self._results.get(self.get_player_option(player.id, 'Army'))
+        results = self._results.get(self.get_player_option(player.id, 'Army'))
+        if not results:
+            return None
+        counts = defaultdict(int)
```
Comment (Wesmania) [inline server/games/game.py]: Plain dict with setdefault would work just as well. Or I guess if we're using collections anyway, we could just use a Counter for the whole thing.

File: server/games/game.py [hunk 2]
```diff
@@ -711,7 +711,14 @@ def get_army_score(self, army):
         return score
 
     def get_army_result(self, player):
-        return self._results.get(self.get_player_option(player.id, 'Army'))
+        results = self._results.get(self.get_player_option(player.id, 'Army'))
+        if not results:
```
Comment (Wesmania) [inline server/games/game.py]: Check if it's None instead plox, it's the usual idiom

File: server/games/game.py (hunk 1 above)
Comment (Rackover) [reply server/games/game.py]: what do you mean by "plain dict with setdefault" ?

Comment (Wesmania) [reply server/games/game.py]: This is a throwaway dict, so I don't see what kind of data you'd need to write into it "later on". You're using a class from collections anyway (defaultdict), so you might as well use the one that's explicitly made for the thing you want to do. If anything, it's less code.

**Pull request merged**